src/main/java/bio/terra/workspace/service/datarepo/DataRepoService.java

```diff
@@ -3,22 +3,39 @@
 import bio.terra.datarepo.api.RepositoryApi;
 import bio.terra.datarepo.client.ApiClient;
 import bio.terra.datarepo.client.ApiException;
+import bio.terra.workspace.app.configuration.DataRepoConfig;
+import bio.terra.workspace.common.exception.ValidationException;
 import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
 @Component
 public class DataRepoService {
 
+  private final DataRepoConfig dataRepoConfig;
+
+  @Autowired
+  public DataRepoService(DataRepoConfig dataRepoConfig) {
+    this.dataRepoConfig = dataRepoConfig;
+  }
+
   private ApiClient getApiClient(String accessToken) {
     ApiClient client = new ApiClient();
     client.setAccessToken(accessToken);
     return client;
   }
 
   private RepositoryApi repositoryApi(String instance, AuthenticatedUserRequest userReq) {
+    validateInstance(instance);
     return new RepositoryApi(getApiClient(userReq.getRequiredToken()).setBasePath(instance));
   }
 
+  public void validateInstance(String instance) {
```

Comment:
I think this should be private - I don't think we expect anything outside this class to call `validateInstance`.

That means we can't test it directly, but testing calls to `snapshotExists` with a bad instance would work.

Reply:
Mentioned to you on Slack, but commenting here to solicit suggestions:
I'm not sure how to test this behavior if this particular method becomes private, because I haven't yet figured out how to mock the dependency on `bio.terra.datarepo.api.RepositoryApi`. Any ideas are welcome!

Reply:
After more slack discussion, we have 4 options:
1. leave the method public + separate
2. move validateInstance to snapshotExists, mock out repositoryApi in the test
3. Add the Powermock library and use that to mock a private method
4. Break validateInstance out into a static method that takes an instances list + instance
(plus probably more we haven't thought of)
I'd lean towards 1 or 2. For 3 I don't like introducing another heavyweight mock library unless we need to, and for 4 we don't expect anything to call this outside of DataRepoService, so a static method feels like the wrong choice.
Open to more thoughts!

Reply:
Option 1 - leaving it public - works for me ... we might even find reason to call this method from elsewhere, in the future

Reply:
Going to leave it as-is for now, then.